### ISSUE TITLE
Using starred expressions in class's subclasses results in error

### ISSUE THREAD

**Issue by Zomatree** [microsoft/pyright#1938]
**Describe the bug**
If you try use starred expressions inside a class's subclasses it results in an error.
```
/home/zomatree/test.py:9:10 - error: Expected class type but received "tuple[Type[Foo], Type[Bar]]" (reportGeneralTypeIssues)
/home/zomatree/test.py:9:9 - error: Argument to class must be a base class (reportGeneralTypeIssues)
  ```
**Expected behavior**
I expect it to be able to use starred expressions inside the class bases

**Screenshots or Code**
![image](https://user-images.githubusercontent.com/39768508/120562336-1b2d5f80-c3fe-11eb-911e-808e4ee976e6.png)

**VS Code extension or command-line**
`pyright 1.1.146`

**Comment by erictraut** [comment]
Yes, this is intended behavior. Static type checkers like pyright and mypy expect you to use simple, static expressions within a class declaration. Mypy emits a similar error in this circumstance.

I'm curious why you want to use starred expressions in this context. I can imagine wanting to use this in cases where you're dynamically building classes, but that sort of dynamic behavior isn't going to work with static type checkers anyway. If these are statically-defined class hierarchies, why not simply declare them using the standard syntax?

**Comment by Rapptz** [comment]
One such use case I had for this was to change the base classes depending on type checking context to aid the static type checker when it came to my dynamic code. Something like:

```py
if TYPE_CHECKING:
    BASES = (FullClass,)
else:
     BASES = (SomeOtherBase,)

@dynamically_fill_attributes
class Test(*BASES):
   ...
```

I tried doing this approach in one of my projects, where I have a `PartialMessage` that is just another form of `Message` except with different requirements. I build the `PartialMessage` by doing some dynamic attribute whitelisting but for most users it's the same as a regular `Message`, so the pseudo inheritance would have helped. I ended up having to reiterate all the attributes I wanted to whitelist ([like so](https://github.com/Rapptz/discord.py/blob/9e4bcd3df7c6139d2e56d6cc7e19efdcb2ae082f/discord/message.py#L1524-L1535)). It wasn't particularly ideal since it was repetitive and even required some `# type: ignore`.

**Comment by mikeshardmind** [comment]
While I don't specifically think this can be generically compatible with static type checking, it appears to me that this should be safe enough as long as whatever is being expanded is `Final[Sequence[Type]]` 

Any issue that would come up if that's the restriction comes down to other type-unsafe behavior. I do see how allowing more than this could present issues, but allowing this narrow variant of it should allow this to remain type-safe, and the safety of any use of this to lie to the type checker to assist it in understanding dynamic code is left on the user.

**Comment by erictraut** [comment]
Thanks for the additional details.

I think this is too esoteric to justify adding special-case logic to a static type checker. I'm going to close this for now. If there are additional upvotes in the future, we could consider adding this support. 

In general, if your code is that dynamic, it's not going to work well with a static type checker. If you find value in static type checking, I recommend looking for alternative approaches in your code.
